src/main/webapp/blur-helpers.js

```diff
@@ -259,3 +273,122 @@ async function loadBlurredPhoto(photo) {
   };
   return getImageWithBlurredAreas(imageDetails, blurRadius);
 }
+
+/**
+ * Function that preloads some of the photos in sessionStorage.
+ * Assigns each photo a value and uses the Knapsack algorithm to decide which
+ * photos to load in maximum CACHE_SIZE space so that the sum of their value is
+ * the biggest.
+*/
+async function preloadPhotos() {
+  // Fetch photos information from the server.
+  const photosResponse = await fetch('/photos');
+  const photos = await photosResponse.json();
+
+  if (photos.length === 0) {
+    return;
+  }
+
+  // Assign each photo a value.
+  const serverTimeResponse = await fetch('server-time');
+  const currentDate = Date.parse(await serverTimeResponse.text());
+  for (const photo of photos) {
+    // We divide by 1000 to convert from miliseconds to seconds.
+    const secondsOld = (currentDate - Date.parse(photo.dateCreated)) / 1000;
+    photo.value = 1 / secondsOld;
+  }
+
+  const photosToSave = await getPhotosToSave(photos);
+
+  // Update sessionStorage with our new photos.
+  sessionStorage.clear();
+  for (const photo of photosToSave) {
+    sessionStorage.setItem(`cache-${photo.id}`, photo.serialized);
+  }
+}
+
+/**
+ * Helper function that uses the Knapsack algorithm to decide which of the
+ * photos to cache to maximize the sum of their values and not exceed the
+ * CACHE_SIZE storage space. Each photo has a value and a sizeInKB variable.
+ * @param {Array<photo>} photos
+ * @return {Array<photo>}
+ */
+async function getPhotosToSave(photos) {
+  const CACHE_SIZE = Math.floor(0.4 * 1024);
```

Comment:
You can specify the units in the name of the variable: CACHE_SIZE_KB/CACHE_SIZE_KiB, or something.

Reply:
Changed it to CACHE_SIZE_KIB, thanks!